quil-py/src/instruction/arithmetic.rs

```diff
@@ -0,0 +1,76 @@
+use quil_rs::instruction::{
+    Arithmetic, ArithmeticOperand, ArithmeticOperator, BinaryLogic, BinaryOperand, BinaryOperands,
+    BinaryOperator,
+};
+
+use rigetti_pyo3::{
+    impl_repr, impl_str, py_wrap_data_struct, py_wrap_type, py_wrap_union_enum,
+    pyo3::{
+        types::{PyFloat, PyInt},
+        Py,
+    },
+};
+
+use super::PyMemoryReference;
+
+py_wrap_union_enum! {
+    PyArithmeticOperator(ArithmeticOperator) as "ArithmeticOperator" {
+        add: Add,
+        subtract: Subtract,
+        divide: Divide,
+        multiply: Multiply
+    }
+}
```

Comment:
My main issue with using a mix of `py_wrap_simple_enum` and `py_wrap_union_enum` is the inconsistent API it exposes. For example, in order to initialize and check equality of an enum variant with `py_wrap_simple_enum` you do something like:

```py
a = MyEnum.Field
assert a == MyEnum.Field
```

Whereas with a `py_wrap_union_enum`, you would do something like:

```py
a = MyEnum.new_a()
assert a.is_a()
```

As a user of the library, I would expect that enums are consistent in their API _or_ there is a good reason as to why they are exposed differently. Because one is "all unit variants" doesn't feel like a justified reason to me, so I opted to make them all with `py_wrap_union_enum!` so they have the same API.

Reply:
Hmm... my expectation in a python environment would be that all-unit-variant enums are the only "real" enums and should have class member constants. I would think that anything which requires creating an instance to only be "enum-like", despite the rust representation

Reply:
I went ahead and used `py_wrap_simple_enum` where it was applicable.